Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -70,9 +74,44 @@ async function run() {
         webDeployPkg = availableWebPackages[0];
 
         var isFolderBasedDeployment = utility.isInputPkgIsFolder(webDeployPkg);
+        var applyFileTransformFlag = JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution;
 
-        if(JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution) {
-            var output = await fileTransformationsUtility.fileTransformations(isFolderBasedDeployment, JSONFiles, xmlTransformation, xmlVariableSubstitution, webDeployPkg);
+        if (applyFileTransformFlag || generateWebConfig) {
+            var folderPath = await utility.generateTemporaryFolderForDeployment(isFolderBasedDeployment, webDeployPkg);
+
+            if (generateWebConfig) {
+
+                //Generate the web.config file if it does not already exist.
+                var webConfigPath = path.join(folderPath, "web.config");
+                if (!fs.existsSync(webConfigPath)) {
+
+                    tl.debug(tl.loc('WebConfigDoesNotExist'));
+
+                    //Extract out the appType parameter as it is not to be replaced.
+                    var webConfigParameters = ParameterParser.parse(webConfigParametersStr);
+                    var appType: string = webConfigParameters['appType'].value;
+                    delete webConfigParameters['appType'];
+
+                    // Get the template path for the given appType
+                    var webConfigTemplatePath = path.join(__dirname, path.normalize('node_modules/webdeployment-common/WebConfigTemplates'), appType.toLowerCase());
```

Comment:
Move this logic to generateWebConfigUtil. The function should take the app name.